pkg/debuginfo/store.go

```diff
@@ -223,7 +223,12 @@ func (s *Store) upload(ctx context.Context, buildID, hash string, r io.Reader) e
 			return status.Error(codes.Internal, err.Error())
 		}
 
-		if err := elfutils.ValidateFile(objFile); err != nil {
+		f, err := elfutils.Open(objFile)
+		if err == nil {
+			err = f.Validate()
+		}
+		if err != nil {
+			f.Close()
```

Comment:
I'd also expect the file is opened only without an error returned from the Open function.